### PR TITLE
[shopsys] support all languages and collations

### LINKED ISSUE #934: Hardcoded language values
### What is happening

I noticed a really weird approach when we received following error:

```
Shopsys\FrameworkBundle\Model\Localization\Exception\InvalidLocaleException

Locale "ro" is not valid
```
I found this:

_vendor/shopsys/framework/src/Model/Localization/Localization.php_
```PHP
class Localization
{
    const DEFAULT_COLLATION = 'en_US';

    /**
     * @var string[]
     */
    protected $languageNamesByLocale = [
        'cs' => 'Čeština',
        'de' => 'Deutsch',
        'en' => 'English',
        'hu' => 'Magyar',
        'pl' => 'Polski',
        'sk' => 'Slovenčina',
    ];

    /**
     * @var string[]
     */
    protected $collationsByLocale = [
        'cs' => 'cs_CZ',
        'de' => 'de_DE',
        'en' => 'en_US',
        'hu' => 'hu_HU',
        'pl' => 'pl_PL',
        'sk' => 'sk_SK',
    ];
```
Why these values are harcoded and this is not part of some config? It really doesn't make a sense, now. 

### Expected result
Great to have some configuration with language settings, to prevent overriding core libraries.

### LINKED ISSUE #1003: Expand instructions how to add new locale in docs
### What is happening
When I want to create a new domain with a new language there are [instructions in the documentation](https://github.com/shopsys/shopsys/blob/master/docs/introduction/how-to-set-up-domains-and-locales.md#36-locale-in-administration).

But if it is a new language like Russia there are no information that I also need to edit [localization.php file](https://github.com/shopsys/shopsys/blob/master/packages/framework/src/Model/Localization/Localization.php). 

### Expected result
In docs, there should be information about the necesity of changes in the Localization.php file when using a new language.

### PR DESCRIPTION
**Pull request by henzigo**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Framework supports only 6 languages. It was relatively difficult to add support for another language. This PR removes those restrictions and starts support all locales including Chinese or Russian language by default. You still have to adjust Demodata and tests.
|New feature| Yes 
|[BC breaks](https://github.com/shopsys/shopsys/blob/master/docs/contributing/backward-compatibility-promise.md)| Yes 
|Fixes issues| closes #1003, closes #934 
|Standards and tests pass| Yes/No
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

I didn't update upgrade notes, because I didn't know what I have to write there.

### CODE REVIEW

**Comment by boris-brtan** [comment]
rebased onto new 8.0

File: packages/framework/src/Component/String/TransformString.php [hunk 1]
```diff
@@ -80,7 +82,10 @@ public static function stringToCamelCase($string)
      */
     protected static function toAscii($string)
     {
-        return iconv('utf-8', 'us-ascii//TRANSLIT//IGNORE', $string);
+        $transliteratorToLatin = Transliterator::create('Any-Latin');
```
[the remark this answers is not in the document]
Comment (henzigo) [reply packages/framework/src/Component/String/TransformString.php]: Yes, with previous implementation Russian and Chinese characters in friendly URLs weren't translated correctly to Latin.

File: project-base/app/config/packages/shopsys_shop.yml [hunk 2]
```diff
@@ -1,6 +1,4 @@
 shopsys_shop:
     router:
-        locale_router_filepaths:
-            cs: '%shopsys.root_dir%/src/Shopsys/ShopBundle/Resources/config/routing_front_cs.yml'
-            en: '%shopsys.root_dir%/src/Shopsys/ShopBundle/Resources/config/routing_front_en.yml'
+        locale_router_filepath_mask: '%shopsys.root_dir%/src/Shopsys/ShopBundle/Resources/config/routing_front_*.yml'
```
[the remark this answers is not in the document]
Comment (henzigo) [reply project-base/app/config/packages/shopsys_shop.yml]: Hello, is there any reason for that change or it's just your preference to use question marks?

You forgot to change: https://github.com/shopsys/shopsys/blob/eb139029a3d3f43f77ff5555df930a6c50e0a339/packages/framework/src/Component/Router/LocalizedRouterFactory.php#L72

[the remark this answers is not in the document]
Comment (boris-brtan) [reply project-base/app/config/packages/shopsys_shop.yml]: i thought that it is path expression, so ?? should mean exactly 2 characters.
what should that factory do?

[the remark this answers is not in the document]
Comment (boris-brtan) [reply project-base/app/config/packages/shopsys_shop.yml]: https://commandbox.ortusbooks.com/usage/parameters/globbing-patterns

[the remark this answers is not in the document]
Comment (henzigo) [reply project-base/app/config/packages/shopsys_shop.yml]: I will not have time to work on that. If you want you can do it by yourself. :)

File: docs/upgrade/UPGRADE-v8.0.0.md [hunk 3]
```diff
@@ -69,6 +69,36 @@ There you can find links to upgrade notes for other versions too.
...
             ```
+### Configuration
+- simplify local configuration ([#1004](https://github.com/shopsys/shopsys/pull/1004))
+    - update `app/config/packages/shopsys_shop.yml`
+        ```diff
+        router:
+        -   locale_router_filepaths:
+        -       cs: '%shopsys.root_dir%/src/Shopsys/ShopBundle/Resources/config/routing_front_cs.yml'
+        -       en: '%shopsys.root_dir%/src/Shopsys/ShopBundle/Resources/config/routing_front_en.yml'
+        +   locale_router_filepath_mask: '%shopsys.root_dir%/src/Shopsys/ShopBundle/Resources/config/routing_front_*.yml'
```
[the remark this answers is not in the document]
Comment (boris-brtan) [reply docs/upgrade/UPGRADE-v8.0.0.md]: now i see that it is just a placeholder

File: project-base/app/config/packages/shopsys_shop.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (boris-brtan) [reply project-base/app/config/packages/shopsys_shop.yml]: now i see that it is just a placeholder

**Comment by boris-brtan** [comment]
>`routing_front_cs.yml` and `routing_front_en.yml` - I do not even see any change of permissions so what does it mean? It can now go to tests 
if you check no white space changes there will write that thera only white space changes

**Comment by vitek-rostislav** [comment]
Oh, now I see these are new empty files in tests, my bad :man_facepalming: :smile:

**Comment by boris-brtan** [comment]
TESTS seems ok also for zh(chinese), rebased on actual 8.0

**Pull request merged**